litert/vendors/intel_openvino/compiler/decoder.h

```diff
@@ -0,0 +1,125 @@
+    /// \brief Get a number of inputs
+    size_t get_input_size() const override { return input_tensor_info_.size(); }
+    /// \brief Get a producer name and its output port index
+    void get_input_node(size_t input_port_idx, std::string &producer_name,
+                        std::string &producer_output_port_name,
+                        size_t &producer_output_port_index) const override {
+        // TODO: Needs implementation ? Benchmark/demo app worked fine even without it.
+        return;
+    }
+    /// \brief Get operation type
+    const std::string &get_op_type() const override { return op_type_; }
+    /// \brief Get node name
+    const std::string &get_op_name() const override { return op_name_; }
+
+    // DecoderBaseOperation Interface implementations :
+    /// \brief Get input tensor name by index
+    std::string get_input_tensor_name(size_t idx) const override {
+        return input_tensor_info_[idx].m_tensor_name;
+    }
+    /// \brief Get input tensor type by index
+    ov::element::Type get_input_tensor_type(size_t idx) const override {
+        return input_tensor_info_[idx].m_element_type;
+    }
+    /// \brief Get output tensor name by index
+    std::string get_output_tensor_name(size_t idx) const override {
+        return output_tensor_info_[idx].m_tensor_name;
+    }
+    /// \brief Get output tensor type by index
+    ov::element::Type get_output_tensor_type(size_t idx) const override {
+        return output_tensor_info_[idx].m_element_type;
+    }
+
+    /// \brief Get input tensor info
+    ov::frontend::tensorflow_lite::TensorMetaInfo get_input_tensor_info(size_t idx) const override {
+        return input_tensor_info_[idx];
+    }
+
+    /// \brief Get output tensor info_
+    ov::frontend::tensorflow_lite::TensorMetaInfo get_output_tensor_info(
+        size_t idx) const override {
+        return output_tensor_info_[idx];
+    }
+
+    /// \brief Get a number of outputs
+    size_t get_output_size() const override { return output_tensor_info_.size(); }
+
+private:
+    std::string op_type_;
+    std::string op_name_;
+    std::vector<ov::frontend::tensorflow_lite::TensorMetaInfo> input_tensor_info_;
+    std::vector<ov::frontend::tensorflow_lite::TensorMetaInfo> output_tensor_info_;
+    const LiteRtOp litert_op_;
+    const LiteRtOpCode litert_op_code_;
+};
+
+class DecoderTensor : public ov::frontend::tensorflow_lite::DecoderBaseTensor {
+public:
+    explicit DecoderTensor(ov::frontend::tensorflow_lite::TensorMetaInfo tensor_meta_info,
+                           int64_t input_index, int64_t output_index)
+        : m_tensor_meta_info(tensor_meta_info),
+          input_index_(input_index),
+          output_index_(output_index){};
+
+    ov::frontend::tensorflow_lite::TensorMetaInfo get_tensor_info() const override {
+        return m_tensor_meta_info;
+    }
+
+    /// \brief Get input index for tensor
+    int64_t get_input_idx() const override { return input_index_; }
```

Comment:
please note google style guide is pascal case for all method/function names

GetInputIdx etc